python_modules/dagster/dagster/_grpc/server.py

```diff
@@ -924,17 +929,25 @@ def GetCurrentRuns(self, request, context) -> api_pb2.GetCurrentRunsReply:
             )
 
 
-def server_termination_target(termination_event, server):
+def server_termination_target(termination_event, server, logger):
     termination_event.wait()
-    # We could make this grace period configurable if we set it in the ShutdownServer handler
-    server.stop(grace=5)
+
+    shutdown_grace_period = default_grpc_server_shutdown_grace_period()
+
+    logger.info(
+        f"Stopping server once all current RPC calls terminate or {shutdown_grace_period} seconds"
+        " pass"
+    )
+    finished_shutting_down_rpcs_event = server.stop(grace=shutdown_grace_period)
+    finished_shutting_down_rpcs_event.wait()
```

Comment:
does this `wait` call need a `timeout` set?

Reply:
I would not have expected it to, but I am trying that now

Reply:
Returns:
A threading.Event that will be set when this Server has completely stopped, i.e. when running RPCs either complete or are aborted and all handlers have terminated.